test.py

```diff
@@ -1,12 +1,201 @@
+            print
+            success = False
+            raise ex
+
+    return success
+
+
+def cleanup(self):
+    global endpoints
+
+    endpoints = get_config()
+    omf_types = get_json_file('OMF-Types.json')
+    omf_containers = get_json_file('OMF-Containers.json')
+
+    # Step 9 - Cleanup
+    print('Deletes')
+    success = True
+    for endpoint in endpoints:
+        try:
+            # delete containers
+            for omf_container in omf_containers:
+                send_message_to_omf_endpoint(
+                    endpoint, 'container', [omf_container], action='delete')
+
+            # delete types
+            for omf_type in omf_types:
+                send_message_to_omf_endpoint(
+                    endpoint, 'type', [omf_type], action='delete')
+
+        except Exception as ex:
+            print(f'Encountered Error: {ex}')
+            print
+            traceback.print_exc()
+            print
+            success = False
+            raise ex
+
+    return success
+
+
+def send_get_request_to_endpoint(endpoint, path='', base=''):
+    '''Sends the get request to the path relative to the base base and returns the response'''
+
+    if base == '':
+        base = endpoint["base-endpoint"]
+
+    # Collect the message headers
+    msg_headers = get_headers(endpoint)
+    msg_headers.pop('omfversion')
+    msg_headers["Accept-Verbosity"] = 'verbose'
+
+    # Send message to base base
+    endpoints_type = endpoint["endpoint-type"]
+    response = {}
+    # If the endpoint is OCS
+    if endpoints_type == EndpointTypes.OCS.value:
+        response = requests.get(
+            base+path,
+            headers=msg_headers,
+            verify=endpoint["verify-ssl"],
+            timeout=endpoint["web-request-timeout-seconds"]
+        )
+    # If the endpoint is EDS
+    elif endpoints_type == EndpointTypes.EDS.value:
+        response = requests.get(
+            base+path,
+            headers=msg_headers,
+            timeout=endpoint["web-request-timeout-seconds"]
+        )
+    # If the endpoint is PI
+    elif endpoints_type == EndpointTypes.PI.value:
+        response = requests.get(
+            base+path,
+            headers=msg_headers,
+            verify=endpoint["verify-ssl"],
+            timeout=endpoint["web-request-timeout-seconds"],
+            auth=(endpoint["username"], endpoint["password"])
+        )
+
+    return(response)
+
+def compare_data(data_format, response, sent_data):
+    '''A helper function for comparing the data returned by either the PI Web API or the SDS'''
+    success = True
+    if data_format == 'SDS':
+        for key in sent_data["values"][0]:
+            if sent_data["values"][0][key] != response[key]:
+                success = False
+    else:
+        split = data_format.split('.')
+        if len(split) == 2:
+            prop = split[1]
+            for key in sent_data["values"][0]:
+                if key == prop and sent_data["values"][0][key] == sent_data:
```

Comment:
is there supposed to be a not equals here somewhere? seems weird that if everything matches it's marking it as not successful. Line 184 has a != but lines 191 and 195 don't.

Reply:
Good catch. I will change that.